src/reachy2_sdk/parts/arm.py

```diff
@@ -423,6 +435,149 @@ def send_cartesian_interpolation(
+        target_rot: Quaternion,
+        arc_direction: str,
+        secondary_radius: Optional[float],
+        nb_steps: int,
+        time_step: float,
+    ) -> None:
+        """Generate elliptical interpolation."""
+        vector_target_origin = target_trans - origin_trans
+
+        center = (origin_trans + target_trans) / 2
+        radius = float(np.linalg.norm(vector_target_origin) / 2)
+
+        vector_origin_center = origin_trans - center
+        vector_target_center = target_trans - center
+
+        if radius == 0:
+            return
+        if secondary_radius is None:
+            secondary_radius = radius
+        if secondary_radius is not None and secondary_radius > 0.3:
+            self._logger.warning("interpolation elliptic_radius was too large, reduced to 0.3")
+            secondary_radius = 0.3
+
+        normal = self._get_normal_vector(vector=vector_target_origin, arc_direction=arc_direction)
+
+        if normal is None:
+            self._logger.warning("arc_direction has no solution. Executing linear interpolation instead.")
+            self._send_linear_interpolation(
+                origin_trans=origin_trans,
+                target_trans=target_trans,
+                origin_rot=origin_rot,
+                target_rot=target_rot,
+                nb_steps=nb_steps,
+                time_step=time_step,
+            )
+            return
+
+        cos_angle = np.dot(vector_origin_center, vector_target_center) / (
+            np.linalg.norm(vector_origin_center) * np.linalg.norm(vector_target_center)
+        )
+        angle = np.arccos(np.clip(cos_angle, -1, 1))
+
+        for t in np.linspace(0, 1, nb_steps):
+            # Interpolated angles
+            theta = t * angle
+
+            # Rotation of origin_vector around the circle center in the plan defined by 'normal'
+            q1 = Quaternion(axis=normal, angle=theta)
+            rotation_matrix = q1.rotation_matrix
+
+            # Interpolated point in plan
+            trans_interpolated = np.dot(rotation_matrix, vector_origin_center)
+            # Adjusting the ellipse
+            ellipse_interpolated = trans_interpolated * np.array([1, 1, secondary_radius / radius])
+            trans_interpolated = ellipse_interpolated + center
+
+            # SLERP pour la rotation
+            q_interpolated = Quaternion.slerp(origin_rot, target_rot, t)
+            rot_interpolated = q_interpolated.rotation_matrix
+
+            # Recompose the interpolated matrix
+            interpolated_matrix = recompose_matrix(rot_interpolated, trans_interpolated)
+
+            request = ArmCartesianGoal(
+                id=self._part_id,
+                goal_pose=Matrix4x4(data=interpolated_matrix.flatten().tolist()),
+            )
+            self._stub.SendArmCartesianGoal(request)
+            time.sleep(time_step)
+
+    def _get_normal_vector(self, vector: npt.NDArray[np.float64], arc_direction: str) -> Optional[npt.NDArray[np.float64]]:
```

Comment:
this function can have offline unit tests :)

Reply:
Added